engine/src/rgbmatrix.cpp

```diff
@@ -700,6 +714,9 @@ void RGBMatrix::updateMapChannels(const RGBMap& map, const FixtureGroup *grp, QL
 
         QVector <quint32> rgb = head.rgbChannels();
         QVector <quint32> cmy = head.cmyChannels();
+        QVector <quint32> white = head.whiteChannels();
```

Comment:
Would it be better to perform these chaching only for their specific color mode?
UpdateMapChannels is performed at each RGBMatrix step: with a lot of fixtures or lot of steps this might be CPU expensive.

Reply:
Yes. I will refactor the method to perform only the necessary calculations. Also, the sanity check for the matrix size (leadting to "continue") will be moved some lines earlier.